Exiled.Events/Patches/Events/Player/ItemDrop.cs

```diff
@@ -33,27 +33,29 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
         {
             List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
 
-            LocalBuilder item = generator.DeclareLocal(typeof(Item));
+            LocalBuilder item = generator.DeclareLocal(typeof(API.Features.Items.Item));
+            LocalBuilder ev = generator.DeclareLocal(typeof(DroppingItemEventArgs));
             Label returnLabel = generator.DefineLabel();
             Label notNullLabel = generator.DefineLabel();
 
             newInstructions.InsertRange(0, new[]
             {
-                // Item item = GetItem(player, itemSerial)
-                // if (item is null)
+                // if (!player.TryGetItem(itemSerial, out Item item)) {
                 //    Handlers.Player.OnDroppingNull(new DroppingNullEventArgs(Player));
                 //    return;
+                // }
                 //
-                // Handlers.Player.OnDroppingNull(new DroppingItemEventArgs(Player, Item, true));
+                // var ev = new DroppingItemEventArgs(player, item, isThrow, true);
+                // Handlers.Player.OnDroppingItem(ev);
                 // if (!ev.IsAllowed)
                 //     return;
+                // isThrow = ev.IsThrown;
                 new(OpCodes.Ldarg_0),
                 new(OpCodes.Ldfld, Field(typeof(InventorySystem.Inventory), nameof(InventorySystem.Inventory._hub))),
                 new(OpCodes.Call, Method(typeof(API.Features.Player), nameof(API.Features.Player.Get), new[] { typeof(ReferenceHub) })),
                 new(OpCodes.Ldarg_1),
-                new(OpCodes.Call, Method(typeof(ItemDrop), nameof(ItemDrop.GetItem))),
-                new(OpCodes.Stloc_S, item.LocalIndex),
-                new(OpCodes.Ldloc_S, item.LocalIndex),
+                new(OpCodes.Ldloca_S, item.LocalIndex),
+                new(OpCodes.Callvirt, FirstMethod(typeof(API.Features.Player), info => info.Name == "TryGetItem" && info.GetCustomAttribute<ObsoleteAttribute>() == null)),
```

Comment:
It's a bad check here cause `TryGetItem` have `ref` parameter

Reply:
It might be possible to do better

Reply:
`new (OpCodes.Callvirt, Method(typeof(API.Features.Player), nameof(API.Features.Player.TryGetItem), new[] { typeof(ushort), typeof(API.Features.Items.Item) },` ?

Reply:
> `new (OpCodes.Callvirt, Method(typeof(API.Features.Player), nameof(API.Features.Player.TryGetItem), new[] { typeof(ushort), typeof(API.Features.Items.Item) },` ?

Tested, not working

Reply:
What's the error, specifically? Invalid opcode during patching or something during runtime? Ideally we should get rid of that extra funky reflection, I know it's possible.
If nothing else we can re-make the helper method that calls TryGetItem() in plain C#, an extra call and stloc is still going to be faster than that reflection used in FirstMethod()

Reply:
No, its cause this method have a `ref` parameter, and I haven't found a way to get this method using `AccessTools.Method`
If there is no normal solution, I will make a helper method

Reply:
Tbh that may be why I ended up making a helper method originally, I probably couldn't figure it out either.

Reply:
Close this review, i found solution